### PR TITLE
Correct validation in build_env.

### LINKED ISSUE #211: [BUG] During `build env` Validation is always performed on default branch
**Describe the bug**
When performing the `open-ce build env` action, validation happens before the BuildTree is constructed, so it clones its own copies of the repo.

**To Reproduce**
Run `build env` with a non default git_tag.

**Expected behavior**
Validation is performed on the specified tag.

### PR DESCRIPTION
**Pull request by bethune-bryant**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [x] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes #211

### CODE REVIEW

**Comment by bethune-bryant** [comment]
@bnemanich 
I had to change it a little after we talked about it.

**Review by npanpaliya — APPROVED** [review]

File: open-ce/validate_config.py [hunk 1]
```diff
@@ -46,19 +46,19 @@ def validate_env_config(conda_build_config, env_config_files, variants, reposito
                                                variant['cudatoolkit'],
                                                repository_folder=repository_folder,
                                                conda_build_config=conda_build_config)
-                validate_build_tree(recipes, variant)
+                validate_build_tree(recipes, recipes.get_external_dependencies(variant))
```
Comment (bnemanich) [inline open-ce/validate_config.py:49]: Do we still need to call validated_build_tree here? It seems like we are calling it twice now. Once in the BuildTree constructor, and once here.

Comment (bethune-bryant) [reply open-ce/validate_config.py:49]: You're right. I didn't commit the change that removed this.
Thanks for catching that!

Comment (bethune-bryant) [reply open-ce/validate_config.py:49]: Actually, this is the call for the validate_config entry point.
The build_tree constructor calls the validate build tree function.

Comment (bnemanich) [reply open-ce/validate_config.py:49]: It looks like BuildTree's constructor will call `validate_config.validate_build_tree`. The constructor is called right before this line. This line then calls `validate_config.validate_build_tree` again.

Comment (bnemanich) [reply open-ce/validate_config.py:49]: The tests seem to be taking twice as long as before as well.

Comment (bethune-bryant) [reply open-ce/validate_config.py]: So the entry_point path was doing a double call now that the validation is happening as part of the build_tree constructor. That should be fixed now.

**Review by bnemanich — APPROVED** [review]

**Pull request merged**